test/addons-napi/7_factory_wrap/myobject.cc

```diff
@@ -0,0 +1,107 @@
+#include "myobject.h"
+
+MyObject::MyObject() : env_(nullptr), wrapper_(nullptr) {}
+
+MyObject::~MyObject() { napi_delete_reference(env_, wrapper_); }
+
+void MyObject::Destructor(void* nativeObject, void* /*finalize_hint*/) {
+  reinterpret_cast<MyObject*>(nativeObject)->~MyObject();
```

Comment:
Fixed in https://github.com/nodejs/node/pull/11975/commits/9c0b1517b894d0a655377d23f9e4b9943d93aa38